rest/config_test.go

```diff
@@ -1204,6 +1204,61 @@ func TestSetupServerContext(t *testing.T) {
 	})
 }
 
+// CBG-1583 - config group IDs EE-only
+func TestConfigGroupIDsValidation(t *testing.T) {
+	error := "customization of config_group_id is only supported in enterprise edition"
+	testCases := []struct {
+		name        string
+		cfgGroupIDs string
+		eeMode      bool
+		expectError bool
+	}{
+		{
+			name:        "No change, CE mode",
+			cfgGroupIDs: persistentConfigDefaultGroupID,
+			eeMode:      false,
+			expectError: false,
+		},
+		{
+			name:        "No change, EE mode",
+			cfgGroupIDs: persistentConfigDefaultGroupID,
+			eeMode:      true,
+			expectError: false,
+		},
+		{
+			name:        "Changed, EE mode",
+			cfgGroupIDs: "testGroup",
+			eeMode:      true,
+			expectError: false,
+		},
+		{
+			name:        "Changed, CE mode",
+			cfgGroupIDs: "testGroup",
+			eeMode:      false,
+			expectError: true,
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			if test.eeMode && !base.IsEnterpriseEdition() {
+				t.Skip("EE mode only test case")
+			}
+			if !test.eeMode && base.IsEnterpriseEdition() {
+				t.Skip("CE mode only test case")
+			}
+
+			sc := StartupConfig{Bootstrap: BootstrapConfig{ConfigGroupID: test.cfgGroupIDs}}
+			err := sc.validate()
+			if test.expectError {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), error)
+			} else if err != nil {
+				assert.NotContains(t, err.Error(), error)
+			}
```

Comment:
Minor point but this seems a bit weird. 
Can we not do:
```suggestion
			} else {
				require.NoError(t, err)
			}
```

Reply:
The validation will probably return an error message unrelated to the config group ID as it returns all the validation error messages at once (as errorMessages). This check is to make sure that if an error is returned, it does not contain the error message related to the config group ID.

Reply:
Fair enough. I think you could avoid that by adding a bootstrap server but I guess its possible future validation could break it.